narwhals/_pandas_like/series_dt.py

```diff
@@ -242,3 +245,55 @@ def truncate(self, every: str) -> PandasLikeSeries:
         return self.with_native(
             self.native.dt.floor(f"{multiple}{UNIT_DICT.get(unit, unit)}")
         )
+
+    def offset_by(self, by: str) -> PandasLikeSeries:
+        import pandas as pd
+
+        from narwhals._arrow.utils import UNITS_DICT
+
+        interval = Interval.parse_no_constraints(by)
+        multiple, unit = interval.multiple, interval.unit
+        if unit in {"y", "q", "mo"}:
+            msg = f"Offsetting by {unit} is not yet supported."
+            raise NotImplementedError(msg)
+        native = self.native
+        if self.implementation.is_cudf():
+            msg = "Not implemented for cuDF."
+            raise NotImplementedError(msg)
+        dtype_backend = get_dtype_backend(native.dtype, self.compliant._implementation)
+        if dtype_backend == "pyarrow":
```

Comment:
from here onwards, it's quite complex - maybe we could make a helper function which does this in `_arrow` somewhere, and we reuse that both here and in `_arrow/series_dt.py`?